### ISSUE TITLE
ByteString.IndexOf throws IndexOutOfRangeException after concating two ByteStrings

### ISSUE THREAD

**Issue by Silv3rcircl3** [akkadotnet/akka.net#1852]
After concating two ByteString's IndexOf throws the exception if the element didn't exist.

Example: 
```csharp           
var b = ByteString.Create(new byte[] {1}) + ByteString.Create(new byte[] {2});
            
b.IndexOf(3);
```

**Comment by wmeints** [comment]
I tried the scenario you suggested and indeed it is out of range.
But I expected that, because you are asking for an index that is larger than the size of the array.

The following scenario works correctly as expected:

```
var b = ByteString.Create(new byte[] { 1 }) + ByteString.Create(new byte[] { 2 });
b.IndexOf(1); // The bytestring is 2 bytes, so this should be valid.
```

**Comment by wmeints** [comment]
Changed things around a bit. I found a better way to fix the problem.
After looking around a bit more, I think that HasNext should return false when there are no more sub iterators left in the list.

@Silv3rcircl3  Can you check if this works better for you?

**Comment by Silv3rcircl3** [comment]
Closed via #1887
